src/methods/get.js

```diff
@@ -94,8 +104,21 @@ export function OrdinaryGet(
   // 1. Assert: IsPropertyKey(P) is true.
   invariant(IsPropertyKey(realm, P), "expected property key");
 
+  let desc;
   // 2. Let desc be ? O.[[GetOwnProperty]](P).
-  let desc = O.$GetOwnProperty(P);
+  if (
+    Receiver instanceof AbstractObjectValue &&
+    (Receiver.kind === "Array.from(A,B,C)" || Receiver.kind === "(A).map(B,C)" || Receiver.kind === "Object.keys(A)")
```

Comment:
We should move this into AbstractObjectValue but ideally it wouldn't be an AbstractObjectValue. Instead it should be a concrete ObjectValue and then it can move into the Get check here. https://github.com/facebook/prepack/blob/master/src/values/ObjectValue.js#L730

The issue is that this can be inherited, shadowed and mutated after extraction.